server/server.js

```diff
@@ -69,42 +86,55 @@ app.post("/api/users", async (req, res) =>  {
             "INSERT INTO users (first_name, last_name, email, image, auth0_sub) VALUES ($1, $2, $3, $4, $5) RETURNING *", [first_name, last_name, email, image, auth0_sub]
         );
 
-        res.json(newUser.rows[0]);
+        res.status(201).json(newUser.rows[0]);
         
     } catch (error) {
-        console.error(error.message);
+        console.error("Error Message:", error.message);
+        res.status(500).json({ message: error.message });
     }
 });
 
+
 //displays all books
 app.get("/api/books", async (req, res) =>  {
     console.log("/api/books")
     
     try {
         const {rows : demo_books} = await db.query('SELECT * FROM demo_api');
-        res.send(demo_books);
+        res.status(200).json(demo_books);
+
     } catch (error) {
         console.error("Error Message!:", error.message);
+        res.status(500).json({ message: error.message });
     }
 
 });
 
-////displays specific books
+
+//displays specific books
 app.get("/api/:id", async (req, res) =>  {
-    
-    try {
 
+    try {
         const { id } = req.params;
-        const {rows: book} = await db.query('SELECT * FROM demo_api WHERE api_id = $1',  [id]);
-        res.send(book);
+        const { rows: book } = await db.query('SELECT * FROM demo_api WHERE api_id = $1', [id]);
+
+        if (book.length === 0) {
+            res.status(404).json({ message: "Book not found" });
+        } else {
+            res.status(200).json(book);
+        }
 
     } catch (error) {
         console.error("Error Message!:", error.message);
+        res.status(500).json({ message: error.message });
     }
 
 });
 
+
+//finds the user id based on the auth0_sub
 async function getUserIdFromSub(userSub) {
```

Comment:
I noticed that getUserIdFromSub is called a few different places. It could be useful to create a middleware.

Reply:
Hmm, let me look into it and see if I can refactor my code. Thank you!